dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/guestbook/SelectGuestBookServiceTest.java

```diff
@@ -0,0 +1,123 @@
+package edu.harvard.iq.dataverse.guestbook;
+
+import edu.harvard.iq.dataverse.dataset.datasetversion.DatasetVersionServiceBean;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.guestbook.Guestbook;
+import io.vavr.control.Option;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class SelectGuestBookServiceTest {
+
+    @InjectMocks
+    private SelectGuestBookService selectGuestBookService;
+
+    @Mock
+    private DatasetVersionServiceBean versionService;
+
+    // -------------------- TESTS --------------------
+
+    @Test
+    public void saveGuestbookChanges_WithRemovedGuestbook() {
+        //given
+        DatasetVersion datasetVersion = preparedDatasetVersion();
+        Clock utcClock = Clock.fixed(Instant.ofEpochMilli(1111111111L), ZoneId.of("UTC"));
+        selectGuestBookService.setSystemTime(utcClock);
+
+        Guestbook oldGuestbook = new Guestbook();
+        datasetVersion.getDataset().setGuestbook(oldGuestbook);
+
+        //when
+        when(versionService.updateDatasetVersion(datasetVersion, true)).thenReturn(datasetVersion.getDataset());
+        Dataset savedDataset = selectGuestBookService.saveGuestbookChanges(datasetVersion,
+                                                                           Option.none(),
+                                                                           Option.of(oldGuestbook));
+
+        //then
+        Assert.assertEquals(utcClock.instant(), savedDataset.getGuestbookChangeTime().get().toInstant());
+        Assert.assertNull(savedDataset.getGuestbook());
+    }
+
+    @Test
+    public void saveGuestbookChanges_WithAddedGuestbook() {
+        //given
+        DatasetVersion datasetVersion = preparedDatasetVersion();
+        Clock utcClock = Clock.fixed(Instant.ofEpochMilli(1111111111L), ZoneId.of("UTC"));
+        selectGuestBookService.setSystemTime(utcClock);
```

Comment:
I would put at least these lines in separate setup method annotated with `@Before` or `@BeforeEach` (similarly in next test class).